sqlglot/dialects/redshift.py

```diff
@@ -60,3 +64,55 @@ class Generator(Postgres.Generator):
             exp.DistStyleProperty: lambda self, e: self.naked_property(e),
             exp.Matches: rename_func("DECODE"),
         }
+
+        def values_sql(self, expression: exp.Values) -> str:
+            """
+            Converts `VALUES...` expression into a series of unions.
+
+            Note: If you have a lot of unions then this will result in a large number of recursive statements to
+            evaluate the expression. You may need to increase `sys.setrecursionlimit` to run and it can also be
+            very slow.
+            """
+            if not isinstance(expression.unnest().parent, exp.From):
+                return super().values_sql(expression)
+            rows = [tuple_exp.expressions for tuple_exp in expression.find_all(exp.Tuple)]
+            selects = []
+            for i, row in enumerate(rows):
+                columns = [
+                    exp.alias_(value, column_name) if i == 0 else value
+                    for value, column_name in zip(row, expression.args["alias"].args["columns"])
+                ]
+                selects.append(exp.Select(expressions=columns))
+            subquery_expression = selects[0]
+            if len(selects) > 1:
+                for select in selects[1:]:
+                    subquery_expression = exp.union(subquery_expression, select, distinct=False)
+            return self.subquery_sql(subquery_expression.subquery(expression.alias))
+
+        def with_properties(self, properties: exp.Properties) -> str:
+            """Redshift doesn't have `WITH` as part of their with_properties so we remove it"""
+            return self.properties(properties, prefix=" ", suffix="")
+
+        def renametable_sql(self, expression: exp.RenameTable) -> str:
+            """Redshift only supports defining the table name itself (not the db) when renaming tables"""
+            target_table = expression.this
+            for arg in target_table.args:
+                if arg != "this":
+                    target_table.set(arg, None)
+            this = self.sql(expression, "this")
+            return f"RENAME TO {this}"
+
+        def datatype_sql(self, expression: exp.DataType) -> str:
+            """
+            Redshift converts the `TEXT` data type to `VARCHAR(255)` by default when people more generally mean
+            VARCHAR of max length which is `VARCHAR(max)` in Redshift. Therefore if we get a `TEXT` data type
+            without precision we convert it to `VARCHAR(max)` and if it does have precision then we just convert
+            `TEXT` to `VARCHAR`.
+            """
+            if expression.this == exp.DataType.Type.TEXT:
+                expression = expression.copy()
+                expression.set("this", exp.DataType.Type.VARCHAR)
+                precision = expression.args.get("expressions")
+                if not precision:
+                    expression.set("expressions", [exp.to_identifier("max")])
```

Comment:
expression.append("expressions", exp.to_identifier("max"))

Reply:
why is this even needed? why not just do a TYPE_MAPPING?

TYPE.FLOAT: "VARCHAR(MAX)"

Reply:
Decided to not do the `TYPE_MAPPING` change